java/vector/src/main/java/org/apache/arrow/vector/TimeStampNanoVector.java

```diff
@@ -111,10 +113,7 @@ public LocalDateTime getObject(int index) {
       return null;
     } else {
       final long nanos = valueBuffer.getLong(index * TYPE_WIDTH);
-      final long millis = java.util.concurrent.TimeUnit.NANOSECONDS.toMillis(nanos);
-      final org.joda.time.LocalDateTime localDateTime = new org.joda.time.LocalDateTime(millis,
-              org.joda.time.DateTimeZone.UTC);
-      return localDateTime;
+      return DateUtility.getLocalDateTimeFromEpochNano(nanos);
```

Comment:
This change here is a change in API. It will break all applications using this getObject API.